lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1423,6 +1403,34 @@ def write_store(
         )
         return rval
 
+    @router.delete(
+        "/api/workflows/{workflow_id}",
+        summary="Add the deleted flag to a workflow.",
+    )
+    def delete_workflow(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        workflow_id: DecodedDatabaseIdField = StoredWorkflowIDPathParam,
+    ):
+        workflow_to_delete = self.workflows_manager.get_stored_workflow(trans, workflow_id)
+        self.workflows_manager.check_security(trans, workflow_to_delete)
+        self.workflows_manager.delete(workflow_to_delete)
```

Comment:
What I would do here is move these 3 lines dealing with the business logic into a `delete` method in the service layer which already has access to the `WorkflowManager`, and the same for the `undelete`.